R/latent_individual.R

```diff
@@ -104,46 +104,38 @@ is_latent_individual <- function(data) {
   inherits(data, "epidist_latent_individual")
 }
 
-#' Check if data has the `epidist_latent_individual` class
+#' Create the model-specific component of an `epidist` custom family
 #'
 #' @param data A `data.frame` containing line list data
 #' @param family Output of a call to `brms::brmsfamily()`
 #' @param ... ...
 #'
-#' @method epidist_family epidist_latent_individual
+#' @method epidist_family_model epidist_latent_individual
 #' @family latent_individual
 #' @export
-epidist_family.epidist_latent_individual <- function(data,
-                                                     family = "lognormal",
-                                                     ...) {
+epidist_family_model.epidist_latent_individual <- function(
+  data, family, ...
+) {
   epidist_validate(data)
-  # allows use of stats::family and strings
-  family <- brms:::validate_family(family)
-  non_mu_links <- family[[paste0("link_", setdiff(family$dpars, "mu"))]]
-  non_mu_bounds <- lapply(
-    family$dpars[-1], brms:::dpar_bounds, family = family$family
-  )
+  family <- .add_dpar_info(family)
```

Comment:
Is there a reason we can't call this in `epidist_family` and so avoid it needing to go in every model?

Reply:
(also if we ned it in every model I think it needs to be exported

Reply:
I didn't like that that involves passing in a non-regular family into `epidist_family_model` (something other than just the regular `brms` family)

Reply:
If that is truly the trade-off we are facing here then I can see going either way

Reply:
I guess I see that but also we have already had to add a class to that output right? If your concern is about the user I am not sure that it is going to be that confusing for them?

Reply:
If you feel strong about it I don't mind moving it, I think I just had some aversion as I was writing the documentation like "oh it's a family object plus some other bits, don't ask about those"

Reply:
I agree but I think it adds a lot of value in terms of reducing duplication which I think is important